packages/shared/lib/services/connection.service.ts

```diff
@@ -765,18 +766,32 @@ class ConnectionService {
         return [...new Set(connections.map((config) => config.connection_id))];
     }
 
-    public async deleteConnection(connection: Connection, providerConfigKey: string, environment_id: number, orchestrator: Orchestrator): Promise<number> {
+    public async deleteConnection({
+        connection,
+        providerConfigKey,
+        environmentId,
+        orchestrator,
+        logContextGetter
+    }: {
+        connection: Connection;
+        providerConfigKey: string;
+        environmentId: number;
+        orchestrator: Orchestrator;
+        logContextGetter: LogContextGetter;
+    }): Promise<number> {
         const del = await db.knex
             .from<Connection>(`_nango_connections`)
             .where({
                 connection_id: connection.connection_id,
                 provider_config_key: providerConfigKey,
-                environment_id,
+                environment_id: environmentId,
                 deleted: false
             })
             .update({ deleted: true, credentials: {}, credentials_iv: null, credentials_tag: null, deleted_at: new Date() });
 
         await syncManager.softDeleteSyncsByConnection(connection, orchestrator);
+        const slackService = new SlackService({ logContextGetter, orchestrator });
+        await slackService.closeOpenNotificationForConnection({ connectionId: connection.id!, environmentId });
```

Comment:
it's there to ensure deletion is consistent across the codebase, once again we would be better off with a queue/event system